src/ipc/ipc4/helper.c

```diff
@@ -235,15 +236,21 @@ int ipc_pipeline_free(struct ipc *ipc, uint32_t comp_id)
 static struct comp_buffer *ipc4_create_buffer(struct comp_dev *src, struct comp_dev *sink,
 					      uint32_t src_queue, uint32_t dst_queue)
 {
-	struct ipc4_base_module_cfg *src_cfg;
+	const struct comp_driver *drv = src->drv;
+	struct ipc4_base_module_cfg src_cfg;
 	struct comp_buffer *buffer = NULL;
 	struct sof_ipc_buffer ipc_buf;
-	int buf_size;
+	int buf_size, ret;
 
-	src_cfg = (struct ipc4_base_module_cfg *)comp_get_drvdata(src);
+	ret = drv->ops.get_attribute(src, COMP_ATTR_BASE_CONFIG, &src_cfg);
+	if (ret < 0) {
+		tr_err(&ipc_tr, "failed to get base config for src %x", dev_comp_id(src));
```

Comment:
refine %x to 0x%x to indicate it is a DEX value

Reply:
fixed